src/generators/clientFileGenerator.ts

```diff
@@ -248,7 +265,7 @@ function writeConstructor(
 
   clientConstructor.addStatements([
     ...operationDeclarationDetails.map(
-      ({ name, typeName }) => `this.${name} = new ${typeName}(this)`
+      ({ name, typeName }) => `this.${name} = new ${typeName}Impl(this)`
```

Comment:
Perhaps there could be reasons in the future to have custom names for those classes so I think it is a good idea to include the name of the class in the `operationDeclarationDetails` instead of hardcoding it as one that has the postfix `Impl`.

Reply:
You can have custom names for classes even now with swagger transforms. The extent of having different names for interface and class name seems to be somewhat far fetched idea. I believe we can settle for Impl.